app/components/login_button_component.rb

```diff
@@ -3,18 +3,41 @@
 class LoginButtonComponent < BaseComponent
   VALID_COLORS = ['primary', 'primary-darker', 'primary-lighter'].freeze
 
-  attr_reader :color, :big, :tag_options
+  attr_reader :color, :big, :width, :height, :tag_options
 
   def initialize(color: 'primary', big: false, **tag_options)
     if !VALID_COLORS.include?(color)
       raise ArgumentError, "`color` #{color}} is invalid, expected one of #{VALID_COLORS}"
     end
 
     @big = big
+    @width = big ? 150 : 100
+    @height = big ? 40 : 25
     @color = color
     @tag_options = tag_options
   end
 
+  def svg
+    Rails.root.join(
+      'app', 'assets', 'images',
+      (color != "primary-darker" ? 'logo.svg' : 'logo-white.svg')
```

Comment:
personally I find positive cases easier to check for first?
```suggestion
      (color == "primary-darker" ? 'logo-white.svg' : 'logo.svg')
```

Reply:
thats fair, it can be confusing